servers/cu/src/domain/lib/loadMessages.js

```diff
@@ -610,60 +610,98 @@ function loadCronMessagesWith ({ loadTimestamp, findBlocks, loadBlocksMeta, save
-                 * on behalf of a process, including spawns.
-                 *
-                 * So we can set From here using the Process tags
-                 * and owner, just like we do for any other message
+                 * Ensure the noSave flag is set, so evaluation does not persist
+                 * this process message
                  */
-                From: mapFrom({ tags: ctx.tags, owner: ctx.owner }),
-                Tags: ctx.tags,
-                Epoch: undefined,
-                Nonce: undefined,
-                Timestamp: ctx.block.timestamp,
-                'Block-Height': ctx.block.height,
-                Cron: false
-              },
-              AoGlobal: {
-                Process: { Id: ctx.id, Owner: ctx.owner, Tags: ctx.tags },
-                Module: { Id: ctx.moduleId, Owner: ctx.moduleOwner, Tags: ctx.moduleTags }
+                noSave: true,
+                ordinate: '^',
+                name: `Process Message ${ctx.id}`,
+                message: {
+                  Id: ctx.id,
+                  Signature: ctx.signature,
+                  Data: ctx.data,
+                  Owner: ctx.owner,
+                  /**
+                   * the target of the process message is itself
+                   */
+                  Target: ctx.id,
+                  Anchor: ctx.anchor,
+                  /**
+                   * Since a process may be spawned from another process,
+                   * the owner may not always be an "end user" wallet,
+                   * but the MU wallet that signed and pushed the spawn.
+                   *
+                   * The MU sets From-Process on any data item it pushes
+                   * on behalf of a process, including spawns.
+                   *
+                   * So we can set From here using the Process tags
+                   * and owner, just like we do for any other message
+                   */
+                  From: mapFrom({ tags: ctx.tags, owner: ctx.owner }),
+                  Tags: ctx.tags,
+                  Epoch: undefined,
+                  Nonce: undefined,
+                  Timestamp: ctx.block.timestamp,
+                  'Block-Height': ctx.block.height,
+                  Cron: false
+                },
+                AoGlobal: {
+                  Process: { Id: ctx.id, Owner: ctx.owner, Tags: ctx.tags },
+                  Module: { Id: ctx.moduleId, Owner: ctx.moduleOwner, Tags: ctx.moduleTags }
+                }
               }
             }
           }
 
           /**
            * Emit the merged stream of Cron and Scheduled Messages
            */
-          for await (const message of $messages) yield message
+          while (true) {
+            const { value, done } = message
+            /**
+             * We're done, so break the loop
+             */
+            if (done) break
+
+            yield value
+            const next = await messages.next()
+            message = next
```

Comment:
Since the iteration of the async iterable performed by this loop does not need references to `next` and `prev`, but only the current message. This block can simplified to just:

```js
while(!message.done) {
  yield message.value
  message = await messages.next()
}
```

my fault for not noticing when putting together the pseudo code.